src/lib/server/links/utils.ts

```diff
@@ -1,15 +1,70 @@
 import QRCode from 'qrcode';
-import { LINKS } from './data';
+import type { Link, ParsedLink } from './types';
 
-export function parseLinkId(link: string): keyof typeof LINKS | undefined {
-  const cleanedLinkName = link
+export function parseLink(url: string): ParsedLink | undefined {
+  const hashBeginIdx = url.lastIndexOf('#');
+  let hashStr = '';
+  if (hashBeginIdx !== -1) {
+    hashStr = url.slice(hashBeginIdx + 1);
+    url = url.slice(0, hashBeginIdx);
+  }
+
+  const queryBeginIdx = url.indexOf('?');
+  let queryStr = '';
+  if (queryBeginIdx !== -1) {
+    queryStr = queryBeginIdx === -1 ? '' : url.slice(queryBeginIdx + 1);
```

Comment:
ternary seem redundant, doesnt if statement on line 14 already guarantee queryBeginIdx not -1?

Reply:
Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/607/commits/2e86f853bca54649c57205844fd33fcd4f0fc22c>.